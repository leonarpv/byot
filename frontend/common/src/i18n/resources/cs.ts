/**
 * Plural keys
 * 0 - singular
 * 1 - 2,3,4
 * 2 - 0 and >5
 */
export default {
  'Login': 'Přihlášení',
  'Username or email': 'Použivatelské jméno nebo email',
  'Enter username or email': 'Zadejte použivatelské jméno nebo email',
  'Enter password': 'Zadejte heslo',
  'Password': 'Heslo',
  'Forgot password?': 'Zapomenuté heslo',
  "Don't have an account?": 'Vytvořit konto',
  'Sign up': 'Registrace',
  'Enter valid email': 'Zadejte platný email',
  'Enter email': 'Zadejte email',
  'Password must be longer than 8 characters': 'Heslo musí mít alespoň 8 znaků',
  'Password must contain at least one special character': 'Heslo musí obsahovat alespoň jeden speciální znak',
  'Confirm password': 'Potvrzení hesla',
  'Passwords are not the same': 'Heslá se neshodují',
  'You have to agree with our terms and conditions': 'Musíte souhlasit se všeobecnými podmínkami',
  'First name': 'Jméno',
  'Last name': 'Přimění',
  'Email': 'Email',
  'I allow this website to collect and store submitted data.': 'Souhlasím se spracovaním osobních údajů.',
  'Already have an account?': 'Již máte konto?',
  'Log in': 'Přihlaste se',
  'Request reset password': 'Požádat o změnu hesla',
  'Reset password': 'Zabudnuté heslo',
  'All rights reserved to': 'Všechna práva vyhrazena pro',
  'Back': 'Zpět',
  'Error while trying to login, please, check username or password':
    'Nastala chyba při přihlašovaní, skontrolujte, prosím, zadané údaje',
  'Log out': 'Odhlásit se',
  'training_0': 'trénink',
  'training_1': 'tréninky',
  'training_2': 'tréninků',
  'video_0': 'video',
  'video_1': 'videa',
  'video_2': 'videí',
  'Enter training name': 'Název tréninku',
  'Add video': 'Přidat video',
  'Save training': 'Uložit trénink',
  'Video label': 'Název videa',
  'Start typing to search or paste URL': 'Vyhledejte nebo vložte URL',
  'My training sets': 'Moje seznamy tréninkov',
  'Nothing found': 'Nic nenalezeno',
  'Type to search': 'Pro vyhledávaní začněte psát',
  'Search': 'Vyhledávaní',
  'Home': 'Domů',
  'Create training set': 'Vytvořit seznam tréninků',
  'My profile': 'Můj profil',
};
