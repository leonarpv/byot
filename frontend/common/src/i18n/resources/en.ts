export default {
  'Login': 'Login',
  'Username or email': 'Username or email',
  'Enter username or email': 'Enter username or email',
  'Enter password': 'Enter password',
  'Password': 'Password',
  'Forgot password?': 'Forgot password?',
  "Don't have an account?": "Don't have an account?",
};
