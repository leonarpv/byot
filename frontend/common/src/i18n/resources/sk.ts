export default {
  'Login': 'Prihlásenie',
  'Username or email': 'Používateľské meno alebo email',
  'Enter username or email': 'Zadajte používateľské meno alebo email',
  'Enter password': 'Zadajte heslo',
  'Password': 'Heslo',
  'Forgot password?': 'Zabudnuté heslo',
  "Don't have an account?": 'Vytvoriť konto',
  'Sign up': 'Registrácia',
  'Enter valid email': 'Zadajte platný email',
  'Enter email': 'Zadajte email',
  'Password must be longer than 8 characters': 'Heslo musí mať aspoň 8 znakov',
  'Password must contain at least one special character': 'Heslo musí obsahovať aspoň jeden špeciálny znak',
  'Confirm password': 'Potvrdenie hesla',
  'Passwords are not the same': 'Heslá sa nezhodujú',
  'You have to agree with our terms and conditions': 'Musíte súhlasiť s všeobecnými podmienkami',
  'First name': 'Meno',
  'Last name': 'Priezvisko',
  'Email': 'Email',
  'I allow this website to collect and store submitted data.': 'Súhlasím so spracovaním osobných údajov.',
};
