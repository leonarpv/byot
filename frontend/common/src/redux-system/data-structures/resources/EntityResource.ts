import {Resource, ResourceState} from './Resource';
import * as _ from 'lodash';

export class EntityResource<T> implements Resource<T> {
  protected _data?: T;
  protected _isLoaded: boolean = false;

  /**
   * @param defaultData is supposed to be just simple structure or primitive (eg. {}, [], null, 0)
   * @param state
   */
  constructor(protected defaultData?: T, public state: ResourceState = ResourceState.IDLE) {
    this._data = _.cloneDeep(this.defaultData);
  }

  public get hasData(): boolean {
    return !!this._data;
  }

  public get isProcessing(): boolean {
    return [ResourceState.LOADING].includes(this.state);
  }

  get data(): T | undefined {
    return this._data;
  }

  setData<D = T | undefined>(value: D) {
    this._isLoaded = true;
    this._data = (value as unknown) as T | undefined;
  }

  get isLoaded(): boolean {
    return this._isLoaded;
  }

  public reset<R extends Resource<T> = EntityResource<T>>(): R {
    return (new EntityResource<T>(this.defaultData) as unknown) as R;
  }
}
