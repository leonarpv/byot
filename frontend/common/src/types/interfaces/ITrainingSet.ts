export {TrainingSet as ITrainingSet} from '../../shared/graphql/ts/types';
