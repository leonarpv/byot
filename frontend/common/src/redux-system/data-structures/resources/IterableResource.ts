import {EntityResource} from './EntityResource';
import {Resource, ResourceState} from './Resource';
import {IList} from '../../../types/interfaces/IList';

export class IterableResource<T> extends EntityResource<T[]> {
  public totalCount: number;
  public appendNewData: boolean;

  constructor(
    defaultData: T[] = [],
    {
      state = ResourceState.IDLE,
      totalCount = defaultData.length,
      appendNewData = true,
    }: {
      state?: ResourceState;
      totalCount?: number;
      appendNewData?: boolean;
    } = {}
  ) {
    super(defaultData, state);
    this.totalCount = totalCount;
    this.appendNewData = appendNewData;
  }

  get hasData(): boolean {
    return this.data!.length > 0;
  }

  setData<D = T[] | IList<any>>(value: D) {
    const data = Array.isArray(value) ? value || [] : ((value as unknown) as IList<any>).entries;
    this._data = this.appendNewData ? [...(this._data || []), ...data] : data;
    this.totalCount = Array.isArray(value)
      ? value.length
      : ((value as unknown) as IList<any>).meta?.totalCount;
  }

  get data(): T[] {
    return this._data || [];
  }

  public reset<R extends Resource<T[]> = IterableResource<T>>(): R {
    return (new IterableResource<T>(this.defaultData, {
      appendNewData: this.appendNewData,
      totalCount: this.defaultData?.length,
    }) as unknown) as R;
  }
}
