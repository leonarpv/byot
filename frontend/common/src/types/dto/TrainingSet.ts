import {ITrainingSet} from '../interfaces/ITrainingSet';
import {IDateTime} from '../interfaces/IDateTime';
import {IUser} from '../interfaces/IUser';
import {IList} from '../interfaces/IList';
import {ITraining} from '../interfaces/ITraining';
import {User} from './User';
import {DateTime} from './DateTime';
import {List} from './List';

export class TrainingSet implements ITrainingSet {
  public createdAt: IDateTime;
  public id: string;
  public label: string;
  public owner: IUser;
  public trainings: IList<ITraining>;
  public updatedAt?: IDateTime;

  constructor({createdAt, id, label, owner, trainings, updatedAt}: ITrainingSet) {
    this.createdAt = createdAt instanceof DateTime ? createdAt : new DateTime(createdAt);
    this.updatedAt = updatedAt && updatedAt instanceof DateTime ? updatedAt : new DateTime(updatedAt);
    this.id = id;
    this.label = label;
    this.owner = owner instanceof User ? owner : new User(owner);
    this.trainings = trainings instanceof List ? trainings : new List(trainings);
  }
}
