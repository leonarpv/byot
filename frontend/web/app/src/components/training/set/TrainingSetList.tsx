import * as React from 'react';
import {Grid} from '@material-ui/core';
import {IterableResource} from '@byot-frontend/common/src/redux-system/data-structures/resources/IterableResource';
import {ITrainingSet} from '@byot-frontend/common/src/types/interfaces/ITrainingSet';
import {InfiniteGridList} from '@byot-frontend/web-common/src/components/functional/infinite-list/InfiniteGridList';
import {TrainingSetListItemSkeleton} from './TrainingSetListItemSkeleton';
import {TrainingSetListItem} from './TrainingSetListItem';

interface Props {
  items: IterableResource<ITrainingSet>;
  onLoadMore: (offset: number, limit: number) => void;
}

export const TrainingSetList: React.FC<Props> = (props: Props) => {
  return (
    <InfiniteGridList
      resource={props.items}
      GridProps={{container: true, spacing: 2}}
      next={props.onLoadMore}
      skeleton={
        <Grid item xs={12} md={6}>
          <TrainingSetListItemSkeleton />
        </Grid>
      }>
      {props.items.data.map(item => (
        <Grid item xs={12} md={6} key={item.id}>
          <TrainingSetListItem trainingSet={item} />
        </Grid>
      ))}
    </InfiniteGridList>
  );
};
