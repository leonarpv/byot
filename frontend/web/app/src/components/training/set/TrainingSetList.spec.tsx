import * as React from 'react';
import {render} from '@testing-library/react';
import {TrainingSetList} from './TrainingSetList';
import {ITrainingSet} from '@byot-frontend/common/src/types/interfaces/ITrainingSet';
import {IterableResource} from '@byot-frontend/common/src/redux-system/data-structures/resources/IterableResource';
import {trainingSet} from '@byot-frontend/common/test/fixtures/dto/TrainingSet';

jest.mock('@byot-frontend/web-common/src/components/functional/infinite-list/InfiniteGridList', () => ({
  __esModule: true,
  InfiniteGridList: ({children}: any) => <div data-testid="infinite-grid-list">{children}</div>,
}));

describe('<TrainingSetList/>', () => {
  it('should render', () => {
    const {container} = render(
      <TrainingSetList items={new IterableResource<ITrainingSet>([trainingSet()])} onLoadMore={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });
});
