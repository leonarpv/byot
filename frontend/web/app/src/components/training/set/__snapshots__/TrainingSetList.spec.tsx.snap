// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrainingSetList/> should render 1`] = `
<div>
  <div
    data-testid="infinite-grid-list"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        aria-disabled="false"
        class="MuiBox-root MuiBox-root-9 MuiButtonBase-root MuiListItem-root makeStyles-root-1 MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart makeStyles-root-8"
        role="button"
        tabindex="0"
        to="/training/set/fd255273-5147-504b-ac68-033d90ff4e17"
      >
        <div
          class="makeStyles-image-2 makeStyles-image-5"
        />
        <div
          class="MuiListItemText-root makeStyles-textRoot-4 makeStyles-textRoot-7 MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-primaryText-3 makeStyles-primaryText-6 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Jutacu emibo cu umusosfu re cuznim reletwaj lojeibe lolmuj tujle tovufmu vofizo nuzi regusoj iwmum busifurih weof.
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              33
               
              training
            </p>
            <span
              class="MuiTypography-root makeStyles-root-10 MuiTypography-caption"
            >
              4 months ago
            </span>
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
  </div>
</div>
`;
